### ISSUE TITLE
backup browser is inaccessible with voiceover

### ISSUE THREAD

**Issue by marrie** [Tarsnap/tarsnap-gui#83]
When a user wants to browse for files and they  hit command O the backup browser is totally inaccessible to voiceover users. I personally cannot make heads or tales of this. QT I heard thugh has some accessibility issues with voiceove. Still though can this please be looked into?

Macbook air, 8 gigs of ram, , 256 gig ssd drive.. Os is El Capitan.

![screenshot of backup browser](https://cloud.githubusercontent.com/assets/3614487/15658947/d25a4508-2677-11e6-8df8-167f57c21a71.png)

**Comment by shinnok** [comment]
marrie,

Thanks for reaching out to us. I'm trying to wrap my head around the Accessibility tools and techniques, so please be patient with me while I try to figure out how we can improve on this.

Quick question, can you navigate any of the ListViews in the application? These include the File Browser you posted, the Archives list and the Jobs list.

Another one, how can you read text that can't be focused on?

**Comment by marrie** [comment]
No actually I couldn't.  There's some wonderful documentation at applevis.com I don't know where exactly  on the site but there are wonderful articles, some linked from apple discussing voiceover. QT has had problems in apple and windows though. I know this as I use other clients with QT5 and even that has issues. I hope the QT devs fix those for all platforms of QT. Anyway I hope that helps someone. I'll look  on applevis's site, unless someone beats me to it. This is a wonderful community of visually impaired people who would love to help you test.

**Comment by shinnok** [comment]
Graham, after some investigation I think we might need to implement QAccessibleInterface, QAccessibleTableInterface for some of the custom widgets used in the FilePicker and Backup/Jobs list views in order to make them browsable/accessible via Voice Over. Testing/Improving Accessibility is quite the undertaking and I've only played with OS X so far. Here are the resources I found most useful so far:
http://doc.qt.io/qt-5/accessible.html
http://doc.qt.io/qt-5/accessible-qwidget.html
https://www.apple.com/voiceover/info/guide/

Besides the commonly agreed usability good practices like sticking to existing widgets, having descriptive tooltips&titles for every element, making sure focus is available for every action and in the proper order, etc.. going one step towards improving accessibility requires some planning. Can you please share your plan or insights you have so far in regards to handling this, since you've assigned both issues to yourself?

**Comment by gperciva** [comment]
Certainly, but I'm leaving for BSDCan tomorrow so today I'm preparing my laptop and packing.  I'll get back to you within a week.

**Comment by marrie** [comment]
Take your time. I really appreciate the staff looking into this. I'm right now using the command line as I need  practice, but yeah can't wait to test once the accessibility has ben improved. If I can't for some reason you can reach out to the applevis community. You should be able to find some who are willing and able to give you constructive feedback.

**Comment by shinnok** [comment]
@marrie Do you think it would help if I would add 2 actions "Add File" and "Add Directory" to the system menu bar and keyboard shortcuts that pop the native Open File dialogs, as an interim solution at least? The reasons I don't have that and use only the custom File Browser are:
1. You can't select both files and directories from the same dialog, thus the need for two separate actions;
2. You can't select multiple directories at the same time;
3. You can't control what's being displayed, in regards to hidden, system, symlink files;
Every OS has a combination of those issues with the native dialogs, including OS X.

![native-file-dialog-osx](https://cloud.githubusercontent.com/assets/544446/16093704/04849480-330b-11e6-8aea-464f96a2b144.png)

**Comment by marrie** [comment]
That could work as an interim solution. I see that actually with many apps. Add files, and add folders, etc. Both of which are separate actions.

**Comment by shinnok** [comment]
See comment https://github.com/Tarsnap/tarsnap-gui/issues/82#issuecomment-308757340.
